### PR TITLE
Add homepage spec.

### PR DESCRIPTION
**Pull request by sneakysneakysneaky**
First pass at homepage specs.

Functionality is largely based on cosm's mockups. As such, I have made some assumptions concerning functionality & UI. Specs may need refined as mockups change, more screen sizes are supported, etc.

### CODE REVIEW

File: doc/specs/homepage.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+## Data model
+
+* Notifications
+  * ID
+  * Title
+  * Type
+  * Timestamp
+  * Body
+
+* Blog / Announcement
```
Comment (ghost) [inline doc/specs/homepage.md]: Part of announcement and news specs

File: doc/specs/homepage.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+**Preconditions:** User has at least one unread notification
+
+**Basic flow:**
+
+1. User clicks the "clear all" icon/button. All visible notifications are marked as read.
+
+## Data model
+
+* Notifications
```
Comment (ghost) [inline doc/specs/homepage.md]: Should be in its own spec, but some redundancy here wont hurt

**Comment by sneakysneakysneaky** [comment]
Homepage spec revised. Removed announcement & news requirements; will be included in those relevant specs. Added personalization requirements.

File: doc/specs/homepage.md [hunk 3]
```diff
@@ -0,0 +1,118 @@
+# Homepage
+
+## Metadata
```
Comment (ghost) [inline doc/specs/homepage.md]: No metadata or changelog.

File: doc/specs/homepage.md [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+
+**Trigger:** User wants to add a section to their homepage
+
+**Preconditions:**
+
+* User has not already added all sections to their homepage.
+
+**Postconditions:**
+
+* Success
```
Comment (ghost) [inline doc/specs/homepage.md]: I don't think "Success" is necessary.

File: doc/specs/homepage.md [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+  * Section is now visible in the user's homepage in the desired column
+
+**Basic flow:**
+
+1. The user clicks the add button/icon in the desired column on the homepage.
+1. The user is asked which section they would like to add. The user selects the desired section.
```
Comment (ghost) [inline doc/specs/homepage.md]: 1. -> 2.

Comment (ghost) [inline doc/specs/homepage.md]: What if a user tries to add the same section twice?

File: doc/specs/homepage.md [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+
+**Trigger:** User wants to remove a section from their homepage.
+
+**Preconditions:**
+
+* User has at least one section already added to their homepage.
+
+**Postconditions:**
+
+* Success
```
Comment (ghost) [inline doc/specs/homepage.md]: See above.

File: doc/specs/homepage.md [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+  * Section is now removed from the homepage
+
+**Basic flow:**
+
+1. The user clicks the remove button/icon on the desired section
+1. The user is asked to confirm their choice.
+
+### Re-order homepage sections:
```
Comment (ghost) [inline doc/specs/homepage.md]: 2.0? 1.3?

File: doc/specs/homepage.md [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+
+**Trigger**: User wants to re-order the sections on their homepage.
+
+**Preconditions:**
+
+* User has at least two sections already added to their homepage.
+
+**Postconditions:**
+
+* Success
```
Comment (ghost) [inline doc/specs/homepage.md]: See above.

**Comment by sneakysneakysneaky** [comment]
More revisions. Fixed typos. Added clarifications.

**Comment by ghost** [comment]
Issue #70

**Pull request closed without merging**